substrate/frame/contracts/Cargo.toml

```diff
@@ -48,20 +48,34 @@ sp-io = { path = "../../primitives/io", default-features = false}
 sp-runtime = { path = "../../primitives/runtime", default-features = false}
 sp-std = { path = "../../primitives/std", default-features = false}
 
+xcm = { package = "staging-xcm", path = "../../../polkadot/xcm", default-features = false}
+xcm-executor = { package = "staging-xcm-executor", path = "../../../polkadot/xcm/xcm-executor", default-features = false}
+pallet-xcm = { path = "../../../polkadot/xcm/pallet-xcm", default-features = false}
+
```

Comment:
same

Reply:
I think these crates are going to move into their own repo (cc @KiChjang ), so that should partially address your concern.

I don't see how we could bring xcm features to SC without bringing in these dependencies though... 🤔

Reply:
Yeah, makes sense.

Not my call, but I would strive to have things in the dep graph be _correct_ before landing this PR. Else the slippery slope to a janky cluster of deps crossing boundaries will likely happen.

Reply:
The reasoning is that even solo chain might eventually want to use xcm, to communicate messages over bridges for example. That's why we want xcm to be part of the config, rather than using  other type of indirection like dispatching runtime call or building ad-hoc SC chain extension, which has been how team have interfaced with XCM so far.

We could feature gate everything but I think this is not necessary as the compiler should just strip pallet_xcm code if you use the default `()` noop trait.

Reply:
Yes XCM is not specific to being a parachain. It is a language to express intent. It will also be used for contract <-> runtime communication. In an ideal world it would also be used inside an transaction instead of the unstable encoded `Call`.

Reply:
so I initially I used tight coupling , but I think having an Xcm configuration trait provide a few advantage:
- I find it a bit more readable to do `T::Xcm::some_xcm_foo(..)` rather than `<T as pallet_xcm::Config>::xcm_foo(..)`
- You will need to replace all `T::` by `<T as Config>` to get the pallet to compile
- We can document the set of xcm operation we want to bring in through this trait
- Even if the vision is to use Xcm for all chains, the upgrade path is slightly easier this way as they don't have to implement `pallet_xcm::Config` straight away. 

Happy to update if you feel strongly that we should use tight coupling on the pallet.